### DISCUSSION TITLE
Improve visualization of the "calories in vs. out" graph by color-coding it based on the meal plan and the proximity to the targets.

### DISCUSSION THREAD

**Discussion by gramapp** [jlai/fitness-dashboard#42] [Ideas]
Some people are working on gaining weight, some are losing, and some are maintaining. Yet the graph remains the same. 

Is it possible to improve visualization like so:

If a person is gaining weight (i.e., the person is working on consuming a caloric surplus), then the "calorie in" bar should be red until the surplus minus 50 calories is reached. Once the person is within +/- 50 calories of the target surplus, the bar should be green. Once the person is 50 calories past the surplus, the bar should be yellow.

If a person is losing weight (i.e., the person is working on staying in a caloric deficit), then the "calorie in" bar should be yellow until 50 calories below the deficit. Once the person is within +/- 50 calories of the target deficit, the bar should be green. Once the person is 50 calories past the deficit, the bar should be red.

If a person is maintaining weight (i.e., the person is working on having calories in = calories out), then the "calorie in" bar should be yellow until it is within +/- 50 calories of the calories spent, at which point the bar should turn green.

The "calorie out" should always be displayed in a consistent, neutral color (e.g., gray).

The target calorie level range, based on the current calories out, could be highlighted as well. This feature was never part of the Fitbit dashboard originally, but it would be a cool addition here

Regarding the Fitbit dashboard, it offered meal plans based on "as you go" and another one based on "typical history." I don't know how this affected the graphs, as I always used the "as you go" option. However, the color of the "calories in" bar may need two schemes to show the target zone - one based on the current calorie output and another based on the historic average calorie output.

If the historic average calorie output is used, it should also be shown on the graph of calorie out.

**Comment by jlai** [comment]
I played around with the API but it doesn't give us the actual calorie target, only the current (depending on time of day and calories burned) calories remaining in the day. So we can't use it to display any historical information on whether the goal was met.

While it would be possible to have a user setting to manually enter the calorie goal, I'd rather not add something that could get out of sync with the food plan goals set in the app, especially given that the deficit/surplus changes as you approach your target weight.

**Reply by gramapp** [reply]
Thank you for looking in to this.
